dialogue-core/src/main/java/com/palantir/dialogue/core/BaseUrl.java

```diff
@@ -222,7 +222,7 @@ public String toString() {
         private static final CharMatcher IS_P_CHAR = UNRESERVED.or(SUB_DELIMS);
         private static final CharMatcher IS_PATH = UNRESERVED.or(SUB_DELIMS).or(CharMatcher.anyOf("/"));
         private static final CharMatcher IS_QUERY_CHAR =
-                CharMatcher.anyOf("=&").negate().and(IS_P_CHAR.or(CharMatcher.anyOf("?/")));
+                IS_P_CHAR.or(CharMatcher.anyOf("/?")).and(CharMatcher.noneOf("=&+"));
```

Comment:
Mind adding a comment describing the flexibility the spec provides around the `+` character here?